hibernate-rx-core/src/test/java/org/hibernate/rx/TestConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package org.hibernate.rx;
+
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+public class TestConfiguration {
+  
+  private static final boolean USE_DOCKER = Boolean.getBoolean("docker");
+  
+  private static final MySQLContainer<?> mysql = new MySQLContainer<>()
+        .withUsername("hibernate-rx")
+        .withPassword("hibernate-rx")
+        .withDatabaseName("hibernate-rx")
+        .withReuse(true);
+  
+  private static final PostgreSQLContainer<?> postgresql = new PostgreSQLContainer<>()
+        .withUsername("hibernate-rx")
+        .withPassword("hibernate-rx")
+        .withDatabaseName("hibernate-rx")
+        .withReuse(true);
+  
+  public static String getPostgreSQLURL() {
+    if (USE_DOCKER) {
+      postgresql.start();
```

Comment:
This does a "lazy start" of the DB containers, so that if I only want to run the MySQL tests I don't need to start the PG container.

I can certainly move the container management to a different class, but are you OK with the lazy container starts? I think lazy starts are nice here because it doesn't impose a particular class hierarchy or require `@BeforeClass` methods on every single test class.

Reply:
`TestConfiguration.getPostgreSQLURL() ` is definitely the wrong place where to put this logic.
One solution I prefer is to have two new classes: `MySQLContainer` and `PostgreSQLContainer`

With two methods (at least):
1 `startIfRequired()` it will start the container if docker is selected
2) `getJdbcUrl()`: return the full jdbc url we need to use

I guess these are similar to the classes in TestContainers but they can work without docker.
start() will only work if the USE_DOCKER variable is set so maybe we can call  or something similar

Also, there is no need to hardcode the URL if you already have the values in a variable, let's try to reuse the TestContainers classes as much as possible. Something like:
```
	public static String getJdbcUrl() {
		if ( USE_DOCKER ) {
			return jdbcUrlWithCredentials( mysql.getJdbcUrl(), mysql );
		}
		else {
			return jdbcUrlWithCredentials( "jdbc:mysql://localhost:3306/" + mysql.getDatabaseName(), mysql );
		}
	}

	private static String jdbcUrlWithCredentials(String jdbcUrl, JdbcDatabaseContainer<?> container) {
		return jdbcUrl + "?user=" + container.getUsername() + "&password=" + container.getPassword();
	}
```

I don't think I'm a fan of using TestContainers for this project but I'm willing to give it a try for now that things are relatively simple.